research-software-directory/RSD-as-a-service
# Edit software and project improvements

# Edit software and project improvements

This PR contains number of related changes for edit software and edit project page.

Closes #404
Closes #463 
Closes #495
Closes #568 

Changes proposed in this pull request:
*   Edit project info section:
    *  The changes are save automatically. Save button is removed. View button is rename to View Page.
    *  Project links order can by changed by dragging the item into desired position
* Edit software - info section: The changes are save automatically. Save button is removed. View button is rename to View Page.
* Edit software - contributors: the order of contributors can be changed by dragging the item into desired position

How to test:
* `docker-compose down --volumes` to remove old version
* `docker-compose build` to build all services
* `docker-compose up` to start solution
* Login as rsd_admin to be able to see all inputs on edit software and edit project pages
* Create new software (you can [use this guide](https://hedgedoc.research.software/optj9GCITZGKvBRU4EhF7Q))
  * manipulate all information in first section and view page. Confirm that all changes are saved properly
  * add contributors to software. Change the position of the contributor and confirm that change is reflected on the software page
* Create new project
  * manipulate all information in the first section and view the page. Confirm that all changes are saved properly.
  * create project links. Change the order of the links and confirm that order is applied on the project page
* Test adding mentions when searching by title. Use "Nearest Neighbours Improves Novelty Detection in Autoencoders" as example. Add this publication multiple times in the project impact and output. Adding should not produce error. 

## Example edit software info page (without save button)
![image](https://user-images.githubusercontent.com/9204081/193755452-64764f7f-cc69-423b-a9c8-0d837b4176a5.png)

## Example contributors page with drag feature to change contributor position
![image](https://user-images.githubusercontent.com/9204081/193755528-b362ea18-cf01-42e4-9a9d-9422f972722b.png)

## Example of project info page without save button and with draggable project links
![image](https://user-images.githubusercontent.com/9204081/193756126-01a2b60f-e9be-48af-b12f-2f8dd9ccfa65.png)

PR Checklist:
